### PR TITLE
Auth framework

### PR DESCRIPTION
**Pull request by nisargjhaveri**
Added `auth_lib` to be used across the application. Finishing yet to be done.

The PR is for review, check if you can find any major loophole in logic.

Login, logout works from `auth/login` and `auth/logout`.

To test add a user in DB with `password` field set to `sha1` of the original password. Then you should be able to login with that username and password.

Again, sorry for the UI.

### CODE REVIEW

**Comment by polybuildr** [comment]
Simply a `sha1` hash of the password is a rather insecure password hashing mechanism. Consider using a more robust alternative such as the `password_hash` and `password_verify` functions.

File: app/libraries/auth_lib.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            return $_SESSION['auth'];
+        }
+        else {
+            return false;
+        }
+    }
+
+    function hash_pass($username, $password) {
+        // TODO: Improve this
+        return sha1($password);
```
Comment (polybuildr) [inline app/libraries/auth_lib.php]: `return password_hash($password);`

**Comment by nisargjhaveri** [comment]
@polybuildr, Thank you for your suggestion. We'll look into it, probably in future!?

**Comment by nisargjhaveri** [comment]
Also, I've updated the PR. Someone, please have a look.

**Comment by imjalpreet** [comment]
Looks Good! :+1:

**Pull request merged**